src/Peers/provider.cpp

```diff
@@ -199,64 +205,35 @@ void Provider::processWorkload() {
     proto.set_training_data_index_filename(indexFile);
     string serialized;
     proto.SerializeToString(&serialized);
-    zmq_sender.send(serialized);
+    ml_zmq_sender.send(serialized);
     cout << "Waiting for processed data..." << endl;
-    auto rcvdData = zmq_receiver.receive();
-
-    // turn the string back into a vector
-    vector<int> data;
-    string num = "";
-    for (int i = 0; i < rcvdData.size(); i++) {
-        if (rcvdData[i] == ',') {
-            data.push_back(stoi(num));
-            num = "";
-        } else {
-            num += rcvdData[i];
-        }
-    }
-    data.push_back(stoi(num));
-    cout << "Processed workload with data: ";
-    for (int i = 0; i < data.size(); i++) {
-        cout << data[i] << " ";
-    }
-    cout << endl;
+    auto rcvdData = ml_zmq_receiver.receive();
+    cout << "Received processed data" << endl;
 
     // store result in taskResponse->trainingdata
-    taskResponse = make_unique<TaskResponse>(data);
+    taskResponse = make_unique<TaskResponse>(rcvdData);
     cout << "Completed assigned workload" << endl;
 }
 
-TaskResponse Provider::aggregateResults(vector<vector<int>> followerData) {
-    vector<int> curr_data = taskResponse->getTrainingData();
+TaskResponse Provider::aggregateResults(vector<string> followerData) {
+    // append leader's data to followerData
+    string curr_data = taskResponse->getTrainingData();
```

Comment:
Maybe we comment this out for now and make it a TODO to undo it later? Because our leader is not doing any work for this demo.

Reply:
we can test it out uncommented. if it creates problems, we can comment out later